### DISCUSSION TITLE
first param will not be removed if it begins with `&` instead of `?`

### DISCUSSION THREAD

**Discussion by uBlock-user** [uBlockOrigin/uBlock-issues#2019] [General] [answered]
Ideally it is expected that first param of a url always be appended with `?`, but seems there's a website which doesn't adhere to this norm.

STR -- 

- Add `||video.laxd.com^$removeparam=ref`
- Navigate to `https://video.laxd.com/a/content/20120714Bpr4rQ1f&ref=videofc2`
- Param doesn't get pruned.

@gorhill

**Comment by Yuki2718** [comment]
The original issue: https://github.com/AdguardTeam/AdguardFilters/issues/111536

**Comment by Yuki2718** [comment]
Makes sense: https://github.com/AdguardTeam/CoreLibs/issues/1600#issuecomment-1054086706
Parameter must start from `?`

**Comment by gorhill** [comment, marked as answer]
Query parameters are introduced with `?`. If there is no `?`, there is no query parameters, and it's only a path, `&` is a valid character in a path.
